Packs/Signum/Integrations/Signum/Signum.py

```diff
@@ -0,0 +1,297 @@
+    """Performs a basic GET request to check if the API is reachable and authentication is successful.
+
+    Args:
+        client: Client object with request
+        username: username
+        password: password
+        *_: Usually demisto.args()
+    NOTE: The username and password are requested to be provided in the SOAP Header
+    Returns:
+        'ok' if test successful.
+
+    Raises:
+        DemistoException: If test failed.
+    """
+    results = client.test_module(username=username, password=password)
+    if results.status_code == 200:
+        return None, None, 'ok'
+    raise DemistoException(f'Test module failed, {results}')
+
+
+@logger
+def list_domain_users_command(client: Client,
+                              domain_id: int,
+                              username: str,
+                              password: str,
+                              simple_view: bool = True) -> tuple[str, dict, Union[list, dict]]:
+    """
+        List Domain Users
+
+    Args:
+        client:
+        domain_id: Domain Identification number
+        username: username
+        password: password
+        simple_view: if "True", strip off prefixes, such as "{urn:.*}" and "{http://.*}", from each dictionary Key name.
+    NOTE: The username and password are requested to be provided in the SOAP Header
+
+    Returns:
+        human readable (markdown format), entry context and raw response
+    """
+
+    simple_view: bool = argToBoolean(simple_view)
+    raw_response: dict = client.list_domain_users(domain_id=domain_id, username=username, password=password)
+    title = f'{INTEGRATION_NAME} - list domain users command'
+    entry_context, human_readable_ec, raw_response = list_domain_users_ec(raw_response=raw_response)
+    context_entry: dict = {
+        f"{INTEGRATION_CONTEXT_NAME}.ListDomainUsers": entry_context
+    }
+
+    human_readable = tableToMarkdown(
+        name=title,
+        t=human_readable_ec,
+        removeNull=True,
+    )
+    return human_readable, context_entry, raw_response
+
+
+''' COMMANDS MANAGER / SWITCH PANEL '''
+
+
+def main():
+
+    params = demisto.params()
+    username = params.get('credentials', {}).get('identifier')
+    password = params.get('credentials', {}).get('password')
+    verify_ssl = not params.get('insecure', False)
+    proxy = params.get('proxy')
+    client = Client(
+        base_url=params.get('url'),
+        verify=verify_ssl,
+        proxy=proxy,
+        auth=(
+            username, password
+        )
+    )
+    command = demisto.command()
+    demisto.debug(f'Command being called is {command}')
+    commands = {
+        'test-module': test_module_command,
+        f'{INTEGRATION_COMMAND_NAME}-list-domain-users': list_domain_users_command,
+    }
+    try:
+        readable_output, outputs, raw_response = commands[command](
+            client=client, username=username, password=password, **demisto.args())
```

Comment:
The client class already maintains the username and password (in self.auth) - why not use it?

Reply:
Good point. I am not aware. Thanks!
Let's leave this as is. It is going to be too many changes. But I will use the properties in the existing class in the next development.